docker_push_latest_if_changed.py

```diff
@@ -1,5 +1,180 @@
 #!/usr/bin/env python3
 import argparse
+import hashlib
+import json
+import subprocess
+import typing
+
+
+class ImageNotFoundError(ValueError):
+    pass
+
+
+class _ImageKey(typing.NamedTuple):
```

Comment:
why underscore prefix ?

Reply:
Not too sure what I was thinking at the time either